src/click/core.py

```diff
@@ -2865,6 +2865,7 @@ def prompt_for_value(self, ctx: Context) -> t.Any:
             show_choices=self.show_choices,
             confirmation_prompt=self.confirmation_prompt,
             value_proc=lambda x: self.process_value(ctx, x),
+            show_default=(False if self.show_default is False else True),
```

Comment:
Why not simply: `show_default=self.show_default`?

Reply:
In this class, `self.show_default` is a `Union[bool, str, None]`, but `termui.prompt` accepts only `bool`. Passing through show_default directly gives a type error, and also appears to do the wrong things for values of type `filenam`

Reply:
Got it. Thanks for the explanation.

Would you mind writing it down as a comment in the code? I think it would help others who read this line of code for the first time.